bindgen/gen_nim.py

```diff
@@ -252,239 +298,141 @@ def extract_ptr_type(s):
-        elif is_2d_array_type(field_type):
-            array_type = extract_array_type(field_type)
-            array_nums = extract_array_nums(field_type)
-            if is_prim_type(array_type):
-                nim_type = as_nim_prim_type(array_type)
-                def_val = type_default_value(array_type)
-            elif is_struct_type(array_type):
-                nim_type = as_nim_struct_type(array_type, prefix)
-                def_val = ".{ }"
-            else:
-                nim_type = "???"
-                def_val = "???"
-            t0 = f"array[{array_nums[0]}, array[{array_nums[1]}, {nim_type}]]"
-            l(f"  {field_name}:{t0}")
-        else:
-            l(f"// FIXME: {field_name}:{field_type};")
+        field_type = check_struct_field_type_override(decl['name'], field['name'], field['type'])
+        l(f"  {field_name}:{as_nim_type(field_type, prefix)}")
     l("")
 
 def gen_consts(decl, prefix):
     l("const")
     for item in decl['items']:
-        l(f"  {trim_prefix(item['name'], prefix)}* = {item['value']}")
+        l(f"  {as_camel_case(trim_prefix(item['name'], prefix), prefix)}* = {item['value']}")
     l("")
 
-def gen_enum(decl, prefix):
+def gen_enum(decl, prefix, bitfield=None):
     item_names_by_value = {}
     value = -1
-    hasForceU32 = False
-    hasExplicitValues = False
+    has_force_u32 = False
+    has_explicit_values = False
     for item in decl['items']:
         itemName = item['name']
         if itemName.endswith("_FORCE_U32"):
-            hasForceU32 = True
+            has_force_u32 = True
         elif itemName.endswith("_NUM"):
             continue
         else:
             if 'value' in item:
-                hasExplicitValues = True
+                has_explicit_values = True
                 value = int(item['value'])
             else:
                 value += 1
             item_names_by_value[value] = as_enum_item_name(item['name']);
-    if hasForceU32:
-        l(f"type {as_nim_enum_type(decl['name'], prefix)}* {{.pure, size:4.}} = enum")
+    enum_name = bitfield if bitfield is not None else decl['name']
+    enum_name_nim = as_nim_type_name(enum_name, prefix)
+    l('type')
+    if has_force_u32:
+        l(f"  {enum_name_nim}* {{.pure, size:sizeof(cint).}} = enum")
     else:
-        l(f"type {as_nim_enum_type(decl['name'], prefix)}* {{.pure.}} = enum")
-    if hasExplicitValues:
+        l(f"  {enum_name_nim}* {{.pure.}} = enum")
+    if has_explicit_values:
         # Nim requires explicit enum values to be declared in ascending order
         for value in sorted(item_names_by_value):
             name = item_names_by_value[value]
-            l(f"  {name} = {value},")
+            l(f"    {name} = {value},")
     else:
         for name in item_names_by_value.values():
-            l(f"  {name},")
+            l(f"    {name},")
+    if bitfield is not None:
+        l(f"  {enum_name_nim}s = set[{enum_name_nim}]")
```

Comment:
Hmm.. this doesn't seem to work. I thought Nim bitfields were interoperable with C..